### ISSUE TITLE
`min`, `max` and `square` don't accept complex arrays

### ISSUE THREAD

**Issue by kohr-h** [cupy/cupy#794]
Numpy has a (somewhat strange) policy of ordering according to real part first, and then breaking ties according to the imaginary part. CuPy currently raises the following error:
```python
>>> a = cupy.array([1j, 2, 3])
>>> cupy.min(a)
TypeError                                 Traceback (most recent call last)
[...]/cupy/core/fusion.py in __call__(self, *args, **kwargs)
    819             return self._numpy_op(*args, **kwargs)
    820         else:
--> 821             return self._cupy_op(*args, **kwargs)
    822 
    823 

[...]/cupy/statistics/order.py in amin(a, axis, out, keepdims, dtype)
     30     """
     31     # TODO(okuta): check type
---> 32     return a.min(axis=axis, dtype=dtype, out=out, keepdims=keepdims)
     33 
     34 

[...]/cupy/core/core.pyx in cupy.core.core.ndarray.min()

[...]/cupy/core/core.pyx in cupy.core.core.ndarray.min()

[...]/cupy/core/reduction.pxi in cupy.core.core.simple_reduction_function.__call__()

[...]/cupy/core/elementwise.pxi in cupy.core.core._guess_routine()

TypeError: Wrong type ((<class 'numpy.complex128'>,)) of arguments for cupy_min
```
I don't know if you want to follow Numpy here, but I guess a more specific error message would be a good thing in that case.

**Comment by kohr-h** [comment]
I added `square` to the list although it's a different kind of function, but it uses the same code.

**Comment by themightyoarfish** [comment]
Is there any progress or workaround for this?

**Comment by okuta** [comment]
We implemented some ufunc for complex (#1281).

**Comment by leofang** [comment]
@kmaehashi this is fixed.
